### PR TITLE
fix(bitfield): support Redis positional '#N' offset syntax

### PR DESCRIPTION
**Pull request by nkroker**
Linked issue #3469 

## Problem                                                                                                                                                         
                                                                                                                                                                     
  `BITFIELD` commands with positional offset `#N` syntax failed in kvrocks. Redis supports `#N` as shorthand for `N * bit_width`, so `INCRBY u16 #0 1` means bit     
  offset `0 * 16 = 0`, `#1` means `16`, etc.                                                                                                                         
                                                                                                                                                                     
  Example that worked in Redis but not kvrocks:                                                                                                                  
  BITFIELD mykey OVERFLOW SAT INCRBY u16 #0 1                                                                                                                        
                                                                                                                                                                     
  ## Root Cause                                                                                                                                                      
                                                                                                                                                                     
  `GetBitOffsetFromArgument` called `ParseInt<uint32_t>` directly on offset string. `"#0"` fails integer parsing — `#` prefix was never handled.                     
                                                                                                                                                                     
  ## Fix                                                                                                                                                             
                                                                                                                                                                 
  In `CommandBitfield::Parse()`, after encoding is parsed (bit width known), detect `#` prefix and expand: `offset = N * encoding.Bits()`. Plain integer offsets
  unchanged.
                                                                                                                                                                     
  ## Test                                                                                                                                                            
                                                                                                                                                                     
  Added integration tests in `tests/gocase/unit/type/bitmap/bitmap_test.go` covering:                                                                                
  - `SET`/`GET` with `#N` across multiple positions                                                                                                              
  - `INCRBY` with `#N`                                                                                                                                               
  - `OVERFLOW SAT INCRBY` with `#N` (original failing case)                                                                                                          
  - `BITFIELD_RO GET` with `#N`

### CODE REVIEW

File: src/VERSION.txt [hunk 1]
```diff
@@ -1 +1 @@
-unstable
\ No newline at end of file
+2.15.0
```
Comment (PragmaTwice) [inline src/VERSION.txt]: Please revert this.

Comment (nkroker) [reply src/VERSION.txt]: Done @PragmaTwice

**Review by jihuayu — COMMENTED** [review]
Good catch. There is just one small issue: `BITFIELD bf SET u16 '#268435456' 1` returns an out-of-range error in Redis, but in this PR it is calculated as offset 0.

Could you also add extra test cases for invalid and boundary `#N` offsets, such as `#`, `#abc`, `#-1`, and a multiplication overflow case?

**Review by jihuayu — APPROVED** [review]

**Review by git-hulk — APPROVED** [review]
